homeassistant/components/sql/sensor.py

```diff
@@ -18,22 +19,26 @@
 CONF_COLUMN_NAME = "column"
 CONF_QUERIES = "queries"
 CONF_QUERY = "query"
+CONF_QUERY_TEMPLATE = "query_template"
 
 
-def validate_sql_select(value):
+def validate_sql(value):
     """Validate that value is a SQL SELECT query."""
-    if not value.lstrip().lower().startswith("select"):
-        raise vol.Invalid("Only SELECT queries allowed")
+    if not value.lstrip().lower().startswith(
+        "select"
+    ) and not value.lstrip().lower().startswith("exec"):
+        raise Exception("Only SELECT or EXEC queries allowed")
     return value
 
 
 _QUERY_SCHEME = vol.Schema(
     {
         vol.Required(CONF_COLUMN_NAME): cv.string,
         vol.Required(CONF_NAME): cv.string,
-        vol.Required(CONF_QUERY): vol.All(cv.string, validate_sql_select),
+        vol.Optional(CONF_QUERY): cv.string,
```

Comment:
Why are you removing validation ?

Reply:
I didn't remove it, I moved it.  Look at line 166.  Since the "query_template" is dynamic in nature, I'm firing the validation before it executes.